AusTraits_tutorial.qmd

```diff
@@ -350,28 +465,96 @@ austraits_with_taxa$traits
+
+How do you take the mean of these trait values?
+
+Do you want to include both data from experiments and plants growing under natural conditions? This information is recorded in the `basis_of_record` column.
+
+The function `austraits::summarise_trait_means()` outputs `location` x `taxon_name` x `dataset_id` means, but is not (*currently*) appropriately updating/removing columns such as `observation_id`, `individual_id`, and `replicates`.
+
+An alternative function we're developing is:
+
+```{r, eval = TRUE, message = FALSE}
+weighted <- austraits_weighted_means(austraits, c("leaf_mass_per_area", "leaf_length"))
+
+weighted
+```
+
+This function first groups values at the site level, then at the taxon level. For trait data sourced from floras, where trait values are documented as a minimum and maximum value, it simply takes the mean of these as an additional mean.
+
+## Plotting data {#plotting_functions}
+
+### Plotting trait distributions
+
+Another way to summarise AusTraits data by trait and determine if AusTraits offers sufficient data coverage for a trait of choice, is to plot the distribution of trait values in AusTraits.
+
+As seen in [`A first look at data`](#exploring), the function `austraits::plot_trait_distribution_beeswarm()` plots trait data by dataset_id:
+
+```{r, eval = TRUE}
+austraits$traits %>% 
+  dplyr::filter(trait_name == "leaf_N_per_dry_mass") %>%
+  dplyr::distinct(taxon_name) %>% nrow()
+
+austraits::plot_trait_distribution_beeswarm(austraits, "leaf_N_per_dry_mass", y_axis_category = "dataset_id")
+```
 
-### Plotting locations
+### Plotting data distribution by location
 
 To plot locations, you need to begin by merging on the latitude & longitude data from austraits\$locations.
 
+The `plot_locations` function plots the selected data, separating data into a series of plots based on the variable name selected. You can separate data based on the values of **any** column within the traits table - including `basis_of_record`, `life_stage`, `value_type` - or higher taxon categories (`genus`, `family`).
+
+For instance, `austraits::plot_locations("trait_name")` will output a separate plot for each `trait_name` within the selected data.
+
+A warning: `austraits::plot_locations()` WILL BE VERY SLOW if you request more than \~20 plots. For instance, do not attempt to generate plots for all traits simultaneously. Always first use extract/filter to just select a narrow range of traits, datasets, or taxa.
+
 #### Plot locations by trait
 
-```{r, eval = FALSE}
+```{r, eval = TRUE, message = FALSE, warning = FALSE}
 
 data <-
   austraits %>% 
      austraits::extract_trait(trait_names = c("leaf_mass_per_area")) %>%
      austraits::join_locations()
 
-data$traits %>% austraits::plot_locations("trait_name")
+data$traits %>%
+  filter(!is.na(`latitude (deg)`)) %>%        # reduce the size of the dataframe by only retaining values with geographic coordinates
```

Comment:
elsewhere you've been writing dplyr::filter

Reply:
fixed